src/feature/dirauth/process_descs.c

```diff
@@ -818,6 +818,15 @@ dirserv_add_extrainfo(extrainfo_t *ei, const char **msg)
     goto fail;
   }
 
+  /* If the extrainfo descriptor contains an internal address, reject it. */
+  if (dirserv_router_has_valid_address(ri) < 0) {
```

Comment:
You're checking the routerinfo here, but you should be checking the ServerTransportListenAddr in the extrainfo?